src/frontends/paddle/src/op/set_value.cpp

```diff
@@ -0,0 +1,106 @@
+    return std::make_shared<default_opset::Select>(mask, res, new_node);
+}
+
+NamedOutputs set_value(const NodeContext& node) {
+    const auto input_node = node.get_input("Input");
+    auto value_node = node.get_input("ValueTensor");
+    PADDLE_OP_CHECK(node, (input_node.get_partial_shape().rank().is_static()), "rank must be static");
+    const auto dims = static_cast<int64_t>(input_node.get_partial_shape().rank().get_length());
+    const auto dtype = input_node.get_element_type();
+    const auto axes = node.get_attribute<std::vector<int64_t>>("axes");
+
+    auto input_shape = std::make_shared<default_opset::ShapeOf>(input_node);
+
+    Output<Node> padding_starts_node, padding_ends_node, starts, ends, steps;
+
+    // get starts ends and steps
+    const auto axes_node = default_opset::Constant::create(element::i64, {axes.size(), 1}, axes);
+    const auto spec_dim_node = std::make_shared<default_opset::GatherND>(input_shape, axes_node);
+    if (node.has_input("StartsTensorList") && node.has_input("StepsTensorList") && node.has_input("EndsTensorList")) {
+        starts = handle_minus_index(node.get_ng_inputs("StartsTensorList"), spec_dim_node);
+        ends = handle_minus_index(node.get_ng_inputs("EndsTensorList"), spec_dim_node);
+        steps = handle_minus_index(node.get_ng_inputs("StepsTensorList"), spec_dim_node);
+    } else if (node.has_attribute("starts") && node.has_attribute("steps") && node.has_attribute("ends")) {
+        starts = handle_minus_index(node.get_attribute<std::vector<int64_t>>("starts"), spec_dim_node);
+        ends = handle_minus_index(node.get_attribute<std::vector<int64_t>>("ends"), spec_dim_node);
+        auto step_vec = node.get_attribute<std::vector<int64_t>>("steps");
+        for (size_t i = 0; i < step_vec.size(); i++)
+            PADDLE_OP_CHECK(node, (step_vec[i] == 1), "Elements of steps must be 1");
+        steps = handle_minus_index(step_vec, spec_dim_node);
+    } else
+        PADDLE_OP_CHECK(node, (false), "Invalid arguments!");
+
+    // get padding starts
+    padding_starts_node =
+        default_opset::Constant::create(element::i64, {static_cast<size_t>(dims)}, std::vector<int64_t>(dims));
+    padding_starts_node = std::make_shared<default_opset::ScatterNDUpdate>(padding_starts_node, axes_node, starts);
+
+    // get padding ends
+    padding_ends_node =
+        default_opset::Constant::create(element::i64, {static_cast<size_t>(dims)}, std::vector<int64_t>(dims));
+    const auto ends_update_node = std::make_shared<default_opset::Subtract>(spec_dim_node, ends);
+    padding_ends_node =
+        std::make_shared<default_opset::ScatterNDUpdate>(padding_ends_node, axes_node, ends_update_node);
+
+    // get target value shape
+    Output<Node> value_shape_update_node = std::make_shared<default_opset::Add>(ends_update_node, starts);
+    value_shape_update_node = std::make_shared<default_opset::Subtract>(spec_dim_node, value_shape_update_node);
```

Comment:
I just wonder can we simply use ```ends -starts ``` to get the ```value_shape_update_node```.
Meanwhile pls update the mapping code with some explanation comments on its process which can help other reviewers to understand your codes easily. Thanks e.g https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/paddle/src/op/slice_ops.hpp#L39

Reply:
> I just wonder if we can simply use ends - starts  to get the value_shape_update_node.

No, we can't.
For example:
```python
A: Tensor  # shape (10, 10, 10)
A[2:5, :, 2:5] = value
```
The starts and ends are `[2, 2]` and `[5, 5]`，and the shape of value can be `(3, 1, 3)` or `(3, 10, 3)`.
So we have to make sure it is broadcasted to the correct shape `(3, 10, 3)` instead of `[5-2, 5-2]` .

Reply:
yes, then we can use ```ScatterNDUpdate``` to update the input shape with ```[5-2, 5-2]```?

Reply:
> yes, then we can use `ScatterNDUpdate` to update the input shape with `[5-2, 5-2]`?

Oh yes, you are right.
It's been so long that I forgot what I wrote. 🤡

Reply:
> I just wonder if we can simply use `ends - starts ` to get the `value_shape_update_node`. Meanwhile pls update the mapping code with some explanation comments on its process which can help other reviewers to understand your codes easily. Thanks e.g https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/paddle/src/op/slice_ops.hpp#L39

done.